### PR TITLE
 DO NOT MERGE: fix PHP build issue with upb textformat

### PR DESCRIPTION
**Pull request by jtattermusch**
Based on #23533 - an attempt to solve the PHP artifact/distribtest issues.

### CODE REVIEW

**Comment by jtattermusch** [comment]
This is an improved/alternative version of https://github.com/grpc/grpc/pull/23614

- doesn't need any preprocessing before building PHP (prepare_pecl_extension.sh), as the build files already include the .upbdefs.c files with a patched name
- just ignore the "upbdefs.h" files, it doesn't seem to be necessary to update them (only the .c files generate the conflicting object files?).

**Comment by jtattermusch** [comment]
Depending on the nature of the PHP build issue, there might be one alternative approach -  currently all the UPB generated files go under the "upb-generated" directory. It would be pretty easy to make all the  .upbdefs.c files end up in a different directory (say "upbdefs-generated"), which may prevent the PHP problem (the upb.c and .upbdef.c files will be in different directory which may avoid the collision) and we would't need to patch the file names of the generated files.

**Comment by jtattermusch** [comment]
In a local experiment (ran with `DOCKERHUB_ORGANIZATION=grpctesting tools/run_tests/task_runner.py  -f php linux artifact`)   the PHP artifact build has succeeded.

File: tools/codegen/core/gen_upb_api.sh [hunk 1]
```diff
@@ -124,3 +124,13 @@ do
     --upb_out=$UPB_OUTPUT_DIR \
     --plugin=protoc-gen-upb=$UPB_PLUGIN
 done
+
+
+# In PHP build Makefile, the files with .upb.c suffix collide .upbdefs.c suffix due to a PHP buildsystem bug.
+# Work around this by changing the generated suffix from ".upbdefs.c" to "_ubpdefs.h".
```
Comment (markdroth) [inline tools/codegen/core/gen_upb_api.sh:130]: I think you mean "_upbdefs.c", not "_upbdefs.h".

**Comment by jtattermusch** [comment]
Superseded by https://github.com/grpc/grpc/pull/23835.

**Pull request closed without merging**